File: 7ki_bo/vue-vben-admin/apps/web-naive/scripts/extract-activity-i18n.mjs
import fs from 'node:fs';
import path from 'node:path';
import { fileURLToPath } from 'node:url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const root = path.join(__dirname, '../src/views/activity');
const pat = /[\u4e00-\u9fff][^\n'"]{0,120}/g;
const strings = new Set();
const fileCounts = {};

function walk(d) {
  for (const f of fs.readdirSync(d)) {
    const p = path.join(d, f);
    if (fs.statSync(p).isDirectory()) {
      walk(p);
    } else if (f.endsWith('.vue')) {
      const c = fs.readFileSync(p, 'utf8');
      const matches = [...c.matchAll(pat)].map((m) => m[0].trim());
      if (matches.length) {
        fileCounts[path.relative(root, p)] = matches.length;
        matches.forEach((s) => strings.add(s));
      }
    }
  }
}

walk(root);
console.log('unique', strings.size);
Object.entries(fileCounts)
  .sort((a, b) => b[1] - a[1])
  .forEach(([f, n]) => console.log(n, f));

fs.writeFileSync(
  path.join(__dirname, 'activity-strings.json'),
  JSON.stringify([...strings].sort(), null, 2),
  'utf8',
);

File: 7ki_bo/vue-vben-admin/apps/web-naive/scripts/extract-activity-i18n.py
"""Extract Chinese strings from activity views for i18n."""
import re
import os
import json

root = os.path.join(os.path.dirname(__file__), '..', 'src', 'views', 'activity')
pattern = re.compile(r'[\u4e00-\u9fff][^\n\'"]{0,120}')

strings = set()
file_counts = {}
for dirpath, _, files in os.walk(root):
    for f in files:
        if not f.endswith('.vue'):
            continue
        path = os.path.join(dirpath, f)
        with open(path, encoding='utf-8') as fh:
            content = fh.read()
        matches = pattern.findall(content)
        chinese = [m.strip() for m in matches if re.search(r'[\u4e00-\u9fff]', m)]
        if chinese:
            rel = os.path.relpath(path, root)
            file_counts[rel] = len(chinese)
            strings.update(chinese)

print('Files:', len(file_counts))
print('Unique strings:', len(strings))
for rel, c in sorted(file_counts.items(), key=lambda x: -x[1]):
    print(f'  {c:4d} {rel}')

out = os.path.join(os.path.dirname(__file__), 'activity-strings.json')
with open(out, 'w', encoding='utf-8') as fh:
    json.dump(sorted(strings), fh, ensure_ascii=False, indent=2)
print('Wrote', out)
